site/en/docs/extensions/mv3/tut_puppeteer-testing/index.md

```diff
@@ -0,0 +1,141 @@
+---
+layout: "layouts/doc-post.njk"
+title: "Tutorial: Testing Chrome Extensions with Puppeteer"
+seoTitle: "Tutorial: Testing Chrome Extensions with Puppeteer"
+date: 2023-10-09
+description: How to write an automated test for Chrome Etensions in Puppeteer.
+---
+
+_For general advice on testing extensions, see
+[Automated testing for Chrome Extensions][automated-testing]._
+
+[Puppeteer][puppeteer] provides a framework for building automated tests of websites, which also
+includes the ability to test Chrome Extensions. These are high-level integration tests that test the
+functionality of a website or extension once it has been built into the final product.
+
+## Before you start
+
+Clone or download the [chrome-extensions-samples][samples-repo] repository. We'll use the history
+API demo in `api-samples/history/showHistory` as our test extension.
+
+You'll also need to install [Node.JS][node] which is the runtime Puppeteer is built on.
+
+## Writing your test
+
+### Step 1: Start your Node.JS project
+
+We need to set up a basic Node.JS project. In a new folder, create a `package.json` file with the
+following:
+
+```js
+{
+  "name": "puppeteer-demo",
+  "version": "1.0"
+}
+```
+
+Similar to an extension's manifest.json file, this file is required by all Node projects.
+
+### Step 2: Install Puppeteer
+
+Run `npm install puppeteer` to add Puppeteer as a dependency. It will be automatically added to your
```

Comment:
```suggestion
Run `npm install puppeteer --save-dev` to add Puppeteer as a dependency. It will be automatically added to your
```

Otherwise dependency won't be added

Reply:
I believe NPM now saves to the main `dependencies` key by default: https://docs.npmjs.com/cli/v8/commands/npm-install#save

I think `dependencies` actually makes sense over `devDependencies` here because Playwright is required to run the tests, not just to build or develop them. That's how I usually think about it anyway (there's admittedly some grey area for local tooling and either works in practice).

Reply:
> I believe NPM now saves to the main dependencies key by default: https://docs.npmjs.com/cli/v8/commands/npm-install#save


TIL

> I think dependencies actually makes sense over devDependencies here because Playwright is required to run the tests, not just to build or develop them. That's how I usually think about it anyway (there's admittedly some grey area for local tooling and either works in practice).

Usually, deps required for running tests is definitely a devDependency (my rule of thumb is always to think about is this going to be required in prod or not). In this particular though, as it is a standalone testing project, I agree that it makes more sense as a normal dependency.